torch/csrc/cuda/Event.cpp

```diff
@@ -0,0 +1,179 @@
+  {(char*)"record", (PyCFunction)THCPEvent_record, METH_O, nullptr},
+  {(char*)"wait", (PyCFunction)THCPEvent_wait, METH_O, nullptr},
+  {(char*)"query", (PyCFunction)THCPEvent_query, METH_NOARGS, nullptr},
+  {(char*)"elapsed_time", (PyCFunction)THCPEvent_elapsed_time, METH_O, nullptr},
+  {(char*)"synchronize",
+    (PyCFunction)THCPEvent_synchronize, METH_NOARGS, nullptr},
+  {(char*)"ipc_handle", (PyCFunction)THCPEvent_ipc_handle, METH_NOARGS, nullptr},
+  {nullptr}
+};
+
+PyTypeObject THCPEventType = {
+  PyVarObject_HEAD_INIT(nullptr, 0)
+  "torch._C._CudaEventBase",             /* tp_name */
+  sizeof(THCPEvent),                     /* tp_basicsize */
+  0,                                     /* tp_itemsize */
+  (destructor)THCPEvent_dealloc,         /* tp_dealloc */
+  0,                                     /* tp_print */
+  0,                                     /* tp_getattr */
+  0,                                     /* tp_setattr */
+  0,                                     /* tp_reserved */
+  0,                                     /* tp_repr */
+  0,                                     /* tp_as_number */
+  0,                                     /* tp_as_sequence */
+  0,                                     /* tp_as_mapping */
+  0,                                     /* tp_hash  */
+  0,                                     /* tp_call */
+  0,                                     /* tp_str */
+  0,                                     /* tp_getattro */
+  0,                                     /* tp_setattro */
+  0,                                     /* tp_as_buffer */
+  Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE, /* tp_flags */
+  nullptr,                                  /* tp_doc */
+  0,                                     /* tp_traverse */
+  0,                                     /* tp_clear */
+  0,                                     /* tp_richcompare */
+  0,                                     /* tp_weaklistoffset */
+  0,                                     /* tp_iter */
+  0,                                     /* tp_iternext */
+  THCPEvent_methods,                     /* tp_methods */
+  0,                                     /* tp_members */
+  THPVariable_properties,                /* tp_getset */
+  0,                                     /* tp_base */
+  0,                                     /* tp_dict */
+  0,                                     /* tp_descr_get */
+  0,                                     /* tp_descr_set */
+  0,                                     /* tp_dictoffset */
+  0,                                     /* tp_init */
+  0,                                     /* tp_alloc */
+  THCPEvent_pynew,                       /* tp_new */
+};
+
+bool THCPEvent_init(PyObject *module) {
+  THCPEventClass = (PyObject*)&THCPEventType;
+  if (PyType_Ready(&THCPEventType) < 0)
+    return false;
+  Py_INCREF(&THCPEventType);
+  PyModule_AddObject(module, "_CudaEventBase", (PyObject *)&THCPEventType);
```

Comment:
Should also check the return value and throw python_error() if necessary here. `PyModule_AddObject` returns `-1` on failure.